app/models/iiif_playlist_manifest_presenter.rb

```diff
@@ -0,0 +1,94 @@
+# Copyright 2011-2023, The Trustees of Indiana University and Northwestern
+#   University.  Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+#   under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+#   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+#   specific language governing permissions and limitations under the License.
+# ---  END LICENSE_HEADER BLOCK  ---
+
+class IiifPlaylistManifestPresenter
+  IIIF_ALLOWED_TAGS = ['a', 'b', 'br', 'i', 'img', 'p', 'small', 'span', 'sub', 'sup'].freeze
+  IIIF_ALLOWED_ATTRIBUTES = ['href', 'src', 'alt'].freeze
+
+  attr_reader :playlist, :items
+
+  def initialize(playlist:, items:)
+    @playlist = playlist
+    @items = items
+  end
+
+  def file_set_presenters
+    # Only return master files that have derivatives to avoid oddities in the manifest and failures in iiif_manifest
+    items.select { |item| item.playlist_item.master_file.derivative_ids.size.positive? }
```

Comment:
Could this be shortened?
```suggestion
    items.select { |item| item.master_file.derivative_ids.size.positive? }
```